miso-web/src/main/webapp/scripts/hot_sample.js

```diff
@@ -0,0 +1,485 @@
+        if (progression.indexOf(targetCategory) > progression.indexOf(sourceCategory)) {
+          startProgression += 1;
+        }
+      }
+      // Now, mark all the appropriate column groups active
+      for (i = startProgression; i <= endProgression && i != -1; i++) {
+        show[progression[i]] = true;
+      }
+      // If we aren't starting or finished with a tissue processing, hide those
+      // columns since we don't really want to show tissue processing unless the
+      // user specifically requested it.
+      if (sourceCategory != 'Tissue Processing' && targetCategory != 'Tissue Processing') {
+        show['Tissue Processing'] = false;
+      }
+      
+      return [
+        {
+          header: 'Sample Alias',
+          data: 'alias',
+          validator : HotUtils.validator.optionalTextNoSpecialChars,
+          type: 'text',
+          unpack: function(sam, flat, setCellMeta) {
+            flat.alias = sam.alias;
+            if (sam.nonStandardAlias) {
+              HotUtils.makeCellNSAlias(setCellMeta);
+            }
+          },
+          pack: function(sam, flat, errorHandler) {
+            sam.alias = flat.alias;
+          },
+          include: true
+        },
+        HotUtils.makeColumnForText('Description', true, 'description', 
+            { validator: HotUtils.validator.optionalTextNoSpecialChars }
+        ),
+        {
+          header: 'Date of receipt',
+          data: 'receivedDate',
+          type: 'date',
+          dateFormat: 'YYYY-MM-DD',
+          datePickerConfig: {
+            firstDay: 0,
+            numberOfMonths: 1
+          },
+          allowEmpty: true,
+          include: config.showReceivedDate,
+          unpack: function (sam, flat, setCellMeta) {
+            flat.receivedDate = sam.receivedDate;
+          },
+          pack: function (sam, flat, errorHandler) {
+            sam.receivedDate = flat.receivedDate;
+          }
+        },
+        HotUtils.makeColumnForText('Matrix Barcode', !Constants.automaticBarcodes, 'identificationBarcode',
+            { validator: HotUtils.validator.optionalTextNoSpecialChars }
+        ),
+        HotUtils.makeColumnForEnum('Sample Type', true, true, 'sampleType', Constants.sampleTypes),
+        HotUtils.makeColumnForText('Sci. Name', true, 'scientificName', 
+            { validator: HotUtils.validator.requiredTextNoSpecialChars }
+        ),
+        {
+          header: 'Project',
+          data: 'projectAlias',
+          type: 'dropdown',
+          source: (function () {
+              var comparator = Constants.isDetailedSample ? 'shortName' : 'id';
+              var label = Constants.isDetailedSample ? 'shortName' : 'name';
+              var projectLabels = (config.projects ? config.projects.sort(Utils.array.standardSort(comparator))
+              .map(function(item) {
+                return item[label];
+              }) : []); // use empty array if projects are not provided
```

Comment:
The check for `config.projects` exists because there's no need to load all the projects during propagate so it should only do the sort if `config.projects` exists. We could indeed put check up in `requestConfiguration` to confirm whether `config.projects && config.create && !config.propagate` and error/dialog if not. Would you like me to do that in this PR or in another?